azure-iot-device/azure/iot/device/provisioning/internal/polling_machine.py

```diff
@@ -0,0 +1,441 @@
+            else:
+                dest = event_data.transition.dest
+            logger.info(
+                "Transition complete.  Trigger={}, Src={}, Dest={}, result={}, error{}".format(
+                    event_data.event.name,
+                    event_data.transition.source,
+                    dest,
+                    str(event_data.result),
+                    str(event_data.error),
+                )
+            )
+
+        self._state_machine = Machine(
+            model=self,
+            states=states,
+            transitions=transitions,
+            initial="disconnected",
+            send_event=True,  # Use event_data structures to pass transition arguments
+            finalize_event=_on_transition_complete,
+            queued=True,
+        )
+
+    def register(self, callback=None):
+        """
+        Register the device with the provisioning service.
+        :param:Callback to be called upon finishing the registration process
+        """
+        logger.info("register called from polling machine")
+        self._register_callback = callback
+        self._trig_register()
+
+    def cancel(self, callback=None):
+        """
+        Cancels the current registration process of the device.
+        :param:Callback to be called upon finishing the cancellation process
+        """
+        logger.info("cancel called from polling machine")
+        self._cancel_callback = callback
+        self._trig_cancel()
+
+    def _initialize_register(self, event_data):
+        logger.info("Initializing the registration process.")
+        self._request_response_provider.subscribe(
+            topic=constant.SUBSCRIBE_TOPIC_PROVISIONING, callback=self._on_subscribe_completed
+        )
+
+    def _send_register_request(self, event_data):
+        """
+        Send the registration request.
+        """
+        logger.info("Sending registration request")
+        self._set_query_timer()
+
+        rid = str(uuid.uuid4())
+
+        self._operations[rid] = constant.PUBLISH_TOPIC_REGISTRATION.format(rid)
+        self._request_response_provider.send_request(
+            rid=rid,
+            topic=constant.PUBLISH_TOPIC_REGISTRATION.format(rid),
+            request=" ",
+            callback=self._handle_register_response_received,
+        )
+
+    def _query_operation_status(self, event_data):
+        """
+        Poll the service for operation status.
+        """
+        logger.info("Querying operation status from polling machine")
+        self._set_query_timer()
+
+        rid = str(uuid.uuid4())
+        result = event_data.args[0].args[0]
+
+        operation_id = result.operation_id
+        self._operations[rid] = constant.PUBLISH_TOPIC_QUERYING.format(rid, operation_id)
+        self._request_response_provider.send_request(
+            rid=rid,
+            topic=constant.PUBLISH_TOPIC_QUERYING.format(rid, operation_id),
+            request=" ",
+            callback=self._handle_query_response_received,
```

Comment:
>self._handle_query_response_received, [](start = 21, length = 37)

any particular reason this isn't a function defined in this block, instead of a convention-private method hanging around on the object itself?

Reply:
changed the name

---
In reply to: [282965963](https://github.com/Azure/azure-iot-sdk-python-preview/pull/81#discussion_r282965963) [](ancestors = 282965963)